### DISCUSSION TITLE
Port order after skrf.connect

### DISCUSSION THREAD

**Discussion by tzortea** [scikit-rf/scikit-rf#1027] [Q&A] [answered]
When using `skrf.connect()`, what is the port order of the result in terms of the input arguments' port order?

E.g. say I have an s4p and an s12p, and I connect them using

`result = skrf.connect(s4p,k,s12p,l,num=2)`

Where should I expect the unconnected ports of the s4p and s12p to end up in the resulting 12-port network?

**Comment by jhillairet** [comment, marked as answer]
There is no obvious way to predict the unconnected port number using the `connect` method with multiport Network. 

However, you can use the following trick to identify the ports numbers : define specific characteristic impedances on a dummy example, so you can track which port have been connect to which:


```python
import skrf as rf

freq = rf.Frequency(1, 1, 1, unit='GHz')
med = rf.DefinedGammaZ0(frequency=freq)
s4p = med.splitter(4, z0=[1,2,3,4])
s6p = med.splitter(6, z0=[11,12,13,14,15,16])
result = rf.connect(s4p, 0, s6p, 0, num=2)
print(result)  # 6-Port Network: '',  1.0-1.0 GHz, 1 pts, z0=[ 3.+0.j  4.+0.j 13.+0.j 14.+0.j 15.+0.j 16.+0.j]
```
in the previous example, one can see that the 1st and 2nd ports of the first Network have been connected to the 1st and 2nd ports of the second Network.

If you want to explicitely control all the connections, it is easier to use [the `Circuit` object](https://scikit-rf.readthedocs.io/en/latest/tutorials/Circuit.html).

**Reply by tzortea** [reply]
Excellent trick, just what I wanted, I wished I had thought of it.  :-)    

Looking through the `connect()` and `connect_s()` and `innerconnect_s()` code, I think there is a way to predict the resulting port ordering, but no need to spend time chasing that down given your elegant solution.
